### PR TITLE
Remove unneeded script parameters in test_cpp_memcheck.sh

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes error introduced in #14992 in `test_cpp_memcheck.sh`
Extra line of parameters removed from the call to `run_cudf_memcheck_ctests.sh`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

File: ci/test_cpp_memcheck.sh [hunk 1]
```diff
@@ -9,7 +9,6 @@ source ./ci/test_cpp_common.sh
 rapids-logger "Memcheck gtests with rmm_mode=cuda"
 
 ./ci/run_cudf_memcheck_ctests.sh \
-    --gtest_output=xml:"${RAPIDS_TESTS_DIR}${test_name}.xml" \
  && EXITCODE=$? || EXITCODE=$?;
```
Comment (vyasr) [inline ci/test_cpp_memcheck.sh:12]: Can we intend this line too? Otherwise the continuation is lost. Alternatively, maybe just merge lines 11 and 12.

**Review by vyasr — CHANGES_REQUESTED** [review]

**Review by vyasr — APPROVED** [review]

**Comment by jakirkham** [comment]
/merge

(adding this proactively so it gets merged once CI passes)

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**